app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-link-tracker.js

```diff
@@ -146,6 +146,10 @@
     hrefIsRelative: function (href) {
       // Checks that a link is relative, but is not a protocol relative url
       return href[0] === '/' && href[1] !== '/'
+    },
+
+    hrefIsAnchor: function (href) {
```

Comment:
Happy with the code as it is (and it's already been merged) but thought I'd raise this as a discussion point. Another approach here would be to invert this function so that it already returns the desired value, instead of returning the value then inverting it, i.e.

```JavaScript
hrefIsNotAnchor: function (href) {
  return href[0] !== '#'
}

if (!isInternalLink && !this.hrefIsRelative(href) && this.hrefIsNotAnchor(href)) {
  //
}
```

I think this would make the code slightly easier to read (and microscopically faster), but interested to know what everyone else thinks.

Reply:
Personally, I find the original easier to read. If we do want to change this though, would we also want to change `!isInternalLink` and `hrefIsRelative` to `isNotInternalLink` and `hrefIsNotRelative` respectively? Otherwise the difference in the checks could potentially get confusing.

Reply:
Agreed (I hadn't checked whether those functions could be as easily inverted though).